### ISSUE TITLE
BUG: Space in directory name causes problems

### ISSUE THREAD

**Issue by hkroger** [metacloud/molecule#783]
# Issue Type

- Bug report

# Molecule and Ansible details

```
$ ansible --version
molecule --version
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides

$ molecule --version
molecule, version 1.20.3
```

- Molecule installation method: brew install molecule
- Ansible installation method: brew install ansible

# Desired Behaviour

I ran `molecule converge` to provision machines. I expected the process to go smoothly and machines to be running after command.

# Actual Behaviour (Bug report only)

Ran `molecule converge` but command failed with following error:

    --> Starting Ansible Run...
    ERROR! Attempted to read ".molecule/ansible_inventory" as YAML: 'AnsibleUnicode' object has no attribute 'keys'
    Attempted to read ".molecule/ansible_inventory" as ini file: .molecule/ansible_inventory:1: Expected key=value host variable assignment, got: Documents/com~apple~CloudDocs/work/ansible/molecule/.vagrant/machines/db1/virtualbox/private_key

I saw that `.molecule/ansible_inventory` had some problematic rows because of spaces in directory names.

One of the problematic rows:
`db1 ansible_ssh_host=127.0.0.1 ansible_ssh_port=2222 ansible_ssh_private_key_file=/Users/hkroger/Library/Mobile Documents/com~apple~CloudDocs/work/ansible/molecule/.vagrant/machines/db1/virtualbox/private_key ansible_ssh_user=vagrant`

**Comment by hkroger** [comment]
At least the issue is different now.

```
$ molecule --debug converge
DEBUG: ANSIBLE ENVIRONMENT
ANSIBLE_CONFIG: .molecule/ansible.cfg
ANSIBLE_FORCE_COLOR: 'true'
ANSIBLE_HOST_KEY_CHECKING: 'false'
ANSIBLE_SSH_ARGS: -o UserKnownHostsFile=/dev/null -o ControlMaster=auto -o ControlPersist=60s
  -o IdentitiesOnly=yes

--> Starting Ansible Run...
DEBUG: COMMAND
/usr/local/bin/ansible-playbook playbook.yml --inventory-file=.molecule/ansible_inventory --connection=ssh --limit=all --user=vagrant --timeout=30 --become --diff
ERROR! Unexpected Exception: 'Host' object has no attribute 'remove_group'
to see the full traceback, use -vvv
```

**Comment by retr0h** [comment]
@hkroger I can't reproduce this current failure.

```
[jodewey:~/Documents/com apple CloudDocs/ansible/foo] [molecule-env-2.7.12] % pwd
/Users/jodewey/Documents/com apple CloudDocs/ansible/foo
```

Any chance you can put your failing role somewhere so I can check it out?

**Comment by alvaroaleman** [comment]
Hey @hkroger,

The issue you are seeing is a bug in Ansible: https://github.com/ansible/ansible/issues/23016
It was introduced in 2.2.2.0 and [will be fixed in 2.2.3.0 once it is out](https://groups.google.com/forum/#!topic/ansible-devel/dmLcxxW_cc4)

**Comment by hkroger** [comment]
Ok, it seems that it was an ansible problem now. Let's close this now and open if it persists after the update is available.
